lms/static/scripts/frontend_apps/components/BasicLtiLaunchApp.js

```diff
@@ -7,16 +7,24 @@ import {
   useState,
 } from 'preact/hooks';
 
-import AuthWindow from '../utils/AuthWindow';
-import { Config } from '../config';
 import { ApiError, apiCall } from '../utils/api';
+import { call as rpcCall } from '../../postmessage_json_rpc/client';
+import { Config } from '../config';
+import { getSidebarWindow } from '../../postmessage_json_rpc/server';
 
+import AuthWindow from '../utils/AuthWindow';
 import Dialog from './Dialog';
 import Button from './Button';
 import ErrorDisplay from './ErrorDisplay';
 import LMSGrader from './LMSGrader';
 import Spinner from './Spinner';
 
+/**
+ * @typedef User
+ * @property {string} userid - Unique user's id
```

Comment:
The `username` => `userid` change here breaks the focused mode switching in the client, presumably until https://github.com/hypothesis/client/pull/1496 lands?

I suggest we change either the client or the server so that we can deploy one part of this without breaking the other.

For example, if you change https://github.com/hypothesis/client/pull/1496/ to accept either `userid` or `username` temporarily, then we can deploy the client change first without breaking Speed Grader or the current Blackboard/Moodle grading implementation, and then separately deploy a change to the LMS app.

Also, I found that the logic which [generates the static config for SpeedGrader](https://github.com/hypothesis/lms/blob/2c888895c922bd999028149e2298c653b3d340ef/lms/views/basic_lti_launch.py#L306) hasn't been updated yet to use `userid`. I presume that will break SpeedGrader?

Reply:
Will fix the client to handle both, then rip it out after we update lms, great idea Rob.